src/prompts.ts

```diff
@@ -1,20 +1,70 @@
-import { FeedbackPrompt } from "./types";
+import { FeedbackPrompt, User } from "./types";
 
 export const parsePromptMessage = (
   message: any,
 ): FeedbackPrompt | undefined => {
   if (
     typeof message?.question !== "string" ||
     !message.question.length ||
-    typeof message?.showAfter !== "number" ||
-    typeof message?.showBefore !== "number"
+    typeof message.showAfter !== "number" ||
+    typeof message.showBefore !== "number" ||
+    typeof message.promptId !== "string" ||
+    !message.promptId.length ||
+    typeof message.featureId !== "string" ||
+    !message.featureId.length
   ) {
     return undefined;
   } else {
     return {
       question: message.question,
       showAfter: new Date(message.showAfter),
       showBefore: new Date(message.showBefore),
+      promptId: message.promptId,
+      featureId: message.featureId,
     };
   }
 };
+
+const rememberMessage = (userId: string, promptId: string) => {
+  localStorage.setItem(`prompt-${userId}`, promptId);
```

Comment:
Can't do `sessionStorage` as it would be cleared if the user closes the tab/window. If the user comes back shortly and the prompt is still "showable", then the user will be presented with the prompt again.

It's either `localStorage` or `cookies`. I think `localStorage` is pretty OK as we'll likely have only one item stored per user's machine at all times. Unless it's a shared machine + account, but even in those cases there are not many downsides. 

Also, speaking from ignorance - would JS-set cookies need to be declared in the "GDPR cookie banner"?

Reply:
In the spirit of GDPR, we actually have to also declare localstorage usage, although I don't think any enforcement has happened on that anywhere yet. So localStorage is not a loophole to GDPR information.

I think we would be bad citizens if we don't clean up after ourselves. If it were our own domain, I would be fine with this trade-off, but in this case we are potentially affecting the reputation of our customers, so I really think we should be looking at sessionStorage or cookies

Reply:
From https://law.stackexchange.com/questions/30739/do-the-gdpr-and-cookie-law-regulations-apply-to-localstorage

Member States shall ensure that the storing of information, or the gaining of access to information already stored, in the terminal equipment of a subscriber or user is only allowed on condition that the subscriber or user concerned has given his or her consent, having been provided with clear and comprehensive information, in accordance with Directive 95/46/EC, inter alia, about the purposes of the processing. 

_**This shall not prevent any technical storage or access for the sole purpose of carrying out the transmission of a communication over an electronic communications network, or as strictly necessary in order for the provider of an information society service explicitly requested by the subscriber or user to provide the service.**_

Reply:
I think the usage of localStorage here is 100% kosher

Reply:
Right, went with @Munter suggestion and switched to cookies instead. I don't think this one requires GDPR notice as it's a user's intent cookie anyway. Please give it a look @Munter. The new file is `prompt-storage.ts`. I separate that to make testing easier as I can mock those without caring about the actual back-end used for storage.

Reply:
Love it :)